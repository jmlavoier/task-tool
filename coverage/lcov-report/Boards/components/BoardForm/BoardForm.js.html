<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Boards/components/BoardForm/BoardForm.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">Boards/components/BoardForm</a> BoardForm.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.33% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/24</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.33% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/24</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
&nbsp;
import style from './BoardForm.sass';
&nbsp;
class BoardForm extends React.Component {
<span class="fstat-no" title="function not covered" >  co</span>nstructor(props) {
<span class="cstat-no" title="statement not covered" >    super(props);</span>
<span class="cstat-no" title="statement not covered" >    this.state = {</span>
      name: props.name,
    };
&nbsp;
<span class="cstat-no" title="statement not covered" >    this.onChangeName = this.onChangeName.bind(this);</span>
<span class="cstat-no" title="statement not covered" >    this.onFieldNameBlur = this.onFieldNameBlur.bind(this);</span>
<span class="cstat-no" title="statement not covered" >    this.onFieldNameKeyup = this.onFieldNameKeyup.bind(this);</span>
<span class="cstat-no" title="statement not covered" >    this.onClickName = this.onClickName.bind(this);</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidMount() {
<span class="cstat-no" title="statement not covered" >    if (this.inputName) {</span>
<span class="cstat-no" title="statement not covered" >      this.inputName.focus();</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentDidUpdate() {
<span class="cstat-no" title="statement not covered" >    if (this.inputName) {</span>
<span class="cstat-no" title="statement not covered" >      this.inputName.focus();</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>FieldNameBlur(event) {
    const { value } = <span class="cstat-no" title="statement not covered" >event.target;</span>
<span class="cstat-no" title="statement not covered" >    this.props.onFieldNameBlur({</span>
      id: this.props.id,
      name: value,
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>ChangeName(event) {
<span class="cstat-no" title="statement not covered" >    this.setState({</span>
      name: event.target.value,
    });
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>FieldNameKeyup(event) {
    const { keyCode } = <span class="cstat-no" title="statement not covered" >event;</span>
    const { value } = <span class="cstat-no" title="statement not covered" >event.target;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (keyCode === 13) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.onFieldNameBlur({</span>
        id: this.props.id,
        name: value,
      });
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>ClickName() {
<span class="cstat-no" title="statement not covered" >    this.props.onClickName(this.props.id);</span>
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
    const { name } = <span class="cstat-no" title="statement not covered" >this.state;</span>
    const { editMode } = <span class="cstat-no" title="statement not covered" >this.props;</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div className={style['board-form']} &gt;
        {editMode ? &lt;input
          type="text"
          ref={<span class="fstat-no" title="function not covered" >(e</span>l) =&gt; { <span class="cstat-no" title="statement not covered" >this.inputName = el; </span>}}
          value={name}
          onChange={this.onChangeName}
          onBlur={this.onFieldNameBlur}
          onKeyUp={this.onFieldNameKeyup}
        /&gt; : &lt;h4&gt;&lt;button type="button" onClick={this.onClickName}&gt;{this.props.name}&lt;/button&gt;&lt;/h4&gt;
        }
      &lt;/div&gt;
    );
  }
}
&nbsp;
BoardForm.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  editMode: PropTypes.bool,
  onFieldNameBlur: PropTypes.func,
  onClickName: PropTypes.func,
};
&nbsp;
BoardForm.defaultProps = {
  name: '',
  editMode: false,
  onFieldNameBlur: <span class="fstat-no" title="function not covered" >()</span> =&gt; {},
  onClickName: <span class="fstat-no" title="function not covered" >()</span> =&gt; {},
};
&nbsp;
export default BoardForm;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Nov 04 2017 19:30:25 GMT-0200 (-02)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
