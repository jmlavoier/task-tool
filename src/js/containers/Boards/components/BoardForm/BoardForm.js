import React from 'react';
import PropTypes from 'prop-types';

import style from './BoardForm.sass';

class BoardForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: props.name
    }
  }

  onChangeName(event) {
    this.setState({
      name: event.target.value,
    })
  }

  onFieldNameBlur(event) {
    const { value } = event.target;
    this.props.onFieldNameBlur({ 
      id: this.props.id,
      name: value,
    });
  }

  componentDidMount() {
    if (this.inputName) {
      this.inputName.focus();
    }
  }

  componentDidUpdate() {
    if (this.inputName) {
      this.inputName.focus();
    }
  }

  onClickName() {
    this.props.onClickName(this.props.id);
  }

  render() {
    const { name } = this.state;
    const { editMode, onFieldNameBlur } = this.props;

    return (
      <div className={style['board-form']} >
        {editMode ? <input type="text" 
                      ref={el => {this.inputName = el}} 
                      value={name} onChange={this.onChangeName.bind(this)} 
                      onBlur={this.onFieldNameBlur.bind(this)} /> 
                  : <h4 onClick={this.onClickName.bind(this)}>{name}</h4>}
      </div>
    );
  }
}

BoardForm.propTypes = {
  name: PropTypes.string,
  editMode: PropTypes.bool,
  onFieldNameBlur: PropTypes.func,
  onClickName: PropTypes.func,
};

BoardForm.defaultProps = {
  name: '',
  editMode: false,
  onFieldNameBlur: () => {},
  onClickName: () => {},
};

export default BoardForm;
