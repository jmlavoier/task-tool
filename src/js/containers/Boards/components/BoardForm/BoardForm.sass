.board-form
  & > input 
    border: 0
    border-bottom: 2px solid #f1f1f1
    background-color: rgba(0,0,0,0.1)
    outline: none
    color: #f1f1f1
    font-size: 14px
    width: 100%