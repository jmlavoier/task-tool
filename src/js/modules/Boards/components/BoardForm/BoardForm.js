import React from 'react';
import PropTypes from 'prop-types';

import style from './BoardForm.sass';

class BoardForm extends React.Component {
  static focusElement(el) {
    el.focus();
  }

  constructor(props) {
    super(props);
    this.state = {
      name: props.name,
    };

    this.onChangeName = this.onChangeName.bind(this);
    this.onFieldNameBlur = this.onFieldNameBlur.bind(this);
    this.onFieldNameKeyup = this.onFieldNameKeyup.bind(this);
    this.onClickName = this.onClickName.bind(this);
  }

  componentDidMount() {
    if (this.inputName) {
      this.constructor.focusElement(this.inputName);
    }
  }

  componentDidUpdate() {
    if (this.inputName) {
      this.constructor.focusElement(this.inputName);
    }
  }

  onFieldNameBlur(event) {
    const { value } = event.target;
    this.props.onFieldNameBlur({
      id: this.props.id,
      name: value,
    });
  }

  onChangeName(event) {
    this.setState({
      name: event.target.value,
    });
  }

  onFieldNameKeyup(event) {
    const { keyCode } = event;
    const { value } = event.target;

    if (keyCode === 13) {
      this.props.onFieldNameBlur({
        id: this.props.id,
        name: value,
      });
    }
  }

  onClickName() {
    this.props.onClickName(this.props.id);
  }

  render() {
    const { name } = this.state;
    const { editMode } = this.props;

    return (
      <div className={style['board-form']} >
        {editMode ? <input
          type="text"
          ref={(el) => { this.inputName = el; }}
          value={name}
          onChange={this.onChangeName}
          onBlur={this.onFieldNameBlur}
          onKeyUp={this.onFieldNameKeyup}
        /> : <h4><button type="button" onClick={this.onClickName}>{this.props.name}</button></h4>
        }
      </div>
    );
  }
}

BoardForm.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string,
  editMode: PropTypes.bool,
  onFieldNameBlur: PropTypes.func,
  onClickName: PropTypes.func,
};

BoardForm.defaultProps = {
  name: '',
  editMode: false,
  onFieldNameBlur: () => {},
  onClickName: () => {},
};

export default BoardForm;
