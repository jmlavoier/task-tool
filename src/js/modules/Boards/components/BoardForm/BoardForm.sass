.board-form
  & > input 
    border: 0
    border-bottom: 2px solid #f1f1f1
    background-color: rgba(0,0,0,0.1)
    outline: none
    color: #f1f1f1
    font-size: 14px
    width: 100%
  
  h4 > button 
    background-color: transparent
    font-weight: bold
    font-size: 16px
    color: #f1f1f1
    border: 0
    outline: none