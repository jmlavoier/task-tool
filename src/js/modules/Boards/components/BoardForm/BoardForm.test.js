import React from 'react';
import { shallow, mount } from 'enzyme';
import BoardForm from './BoardForm';

import uid from 'uid';


test('Should component render with default props', () => {
  const wrapper = shallow(<BoardForm
    editMode={false}
    id={uid()}
    name='Testing'
  />);

  expect(wrapper).toMatchSnapshot();

});

test('Should component render input on edit mode', () => {
  const wrapper = shallow(<BoardForm
    editMode={true}
    id={uid()}
    name='Testing'
  />);

  expect(wrapper).toMatchSnapshot();
});

test('Should component calls onClickName prop on click', () => {
  const onClickName = jest.fn();
  const id = uid();
  
  const wrapper = shallow(<BoardForm
    editMode={false}
    id={id}
    name="Testing"
    onClickName={onClickName}
  />);
  expect(wrapper).toMatchSnapshot();

  wrapper.find('button').simulate('click');
  expect(onClickName).toHaveBeenCalled();
  expect(onClickName.mock.calls[0][0]).toEqual(id);
});