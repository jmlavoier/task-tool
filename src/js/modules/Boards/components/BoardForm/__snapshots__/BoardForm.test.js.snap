// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should component calls onClickName on title click 1`] = `
<div
  className="board-form"
>
  <h4>
    <button
      onClick={[Function]}
      type="button"
    >
      Testing
    </button>
  </h4>
</div>
`;

exports[`Should component render input on edit mode 1`] = `
<div
  className="board-form"
>
  <input
    onBlur={[Function]}
    onChange={[Function]}
    onKeyUp={[Function]}
    type="text"
    value="Testing"
  />
</div>
`;

exports[`Should component render with default props 1`] = `
<div
  className="board-form"
>
  <h4>
    <button
      onClick={[Function]}
      type="button"
    >
      Testing
    </button>
  </h4>
</div>
`;
